commands/mod.js

```diff
@@ -1,31 +1,39 @@
 //Make an announcement
 exports.announcement = function(message, args){
   var channelIdentifier = args.shift();
+  var channel;
   //Check if we are given a channelName or a channelID
+  logger.info(`Attempting to locate channel ${channelIdentifier}`);
   if(channelIdentifier.match(/^[0-9]+$/) != null){
     channel = global.util.getChannel({id: channelIdentifier, type: 'text'});
   }
+  else if(channelIdentifier.match(/^<#[0-9]+>$/) != null){
+    var id_rx = /^<#([0-9]+)>$/g;
+    var id = id_rx.exec(channelIdentifier)[1];
+    channel = global.util.getChannel({id: id, type: 'text'});
+  }
   else{
-    channel = global.util.getChannel({channelName: channelIdentifier, type: 'text'});
+    channel = undefined;
   }
-
+  
   //Channel not found
   if(channel===undefined){
     message.channel.send(`The channel  \`${channelIdentifier}\' wasn't found... please check your spelling!`)
   }
   //Only one channel found
-  if(channel.length===undefined){
+  else if(channel.length===undefined){
+    logger.info(`Sending an announcement to ${channel.name} on behalf of ${message.author.username}`);

```

Comment:
Can we change this ${channel.name} to just ${channel} so it will link the channel?